relayer/chains/cosmos/query.go

```diff
@@ -658,6 +690,62 @@ func (cc *CosmosProvider) QueryUnreceivedPackets(ctx context.Context, height uin
 	return res.Sequences, nil
 }
 
+func sendPacketQuery(channelID string, portID string, seq uint64) []string {
+	return []string{
+		fmt.Sprintf("%s.packet_src_channel='%s'", spTag, channelID),
+		fmt.Sprintf("%s.packet_src_port='%s'", spTag, portID),
+		fmt.Sprintf("%s.packet_sequence='%d'", spTag, seq),
+	}
+}
+
+func writeAcknowledgementQuery(channelID string, portID string, seq uint64) []string {
+	return []string{
+		fmt.Sprintf("%s.packet_dst_channel='%s'", waTag, channelID),
+		fmt.Sprintf("%s.packet_dst_port='%s'", waTag, portID),
+		fmt.Sprintf("%s.packet_sequence='%d'", waTag, seq),
+	}
+}
+
+func (cc *CosmosProvider) QuerySendPacket(
+	ctx context.Context,
+	srcChanID,
+	srcPortID string,
+	sequence uint64,
+) (provider.PacketInfo, error) {
+	ibcMsgs, err := cc.queryIBCMessages(ctx, cc.log, 1, 1000, sendPacketQuery(srcChanID, srcPortID, sequence))
+	if err != nil {
+		return provider.PacketInfo{}, err
+	}
+	for _, msg := range ibcMsgs {
+		if pi, ok := msg.info.(*packetInfo); ok {
+			if pi.Sequence == sequence {
+				return provider.PacketInfo(*pi), nil
+			}
+		}
+	}
+	return provider.PacketInfo{}, fmt.Errorf("no ibc messages found for send_packet query")
```

Comment:
Would it be worth including the query content in the error message?

Reply:
yes, done